File: backend/src/WorkflowHub.Application/Auth/Models/AuthModels.cs
namespace WorkflowHub.Application.Auth.Models;

public sealed record TokenPair(string AccessToken, string RefreshToken);

public sealed record AuthUserDto(
    Guid Id,
    string Email,
    string Name,
    string? DisplayName,
    string? Username,
    string? Role,
    string? Team,
    string? Bio,
    string? AvatarUrl,
    DateTime CreatedAtUtc);

public sealed record AuthResult(TokenPair Tokens, AuthUserDto User);

File: frontend/src/app/core/auth/auth.models.ts
export interface User {
  id: string;
  email: string;
  name: string;
  displayName: string | null;
  username: string | null;
  role: string | null;
  team: string | null;
  bio: string | null;
  avatarUrl: string | null;
  createdAtUtc: string;
}

export interface TokenPair {
  accessToken: string;
  refreshToken: string;
}

export interface AuthResult {
  tokens: TokenPair;
  user: User;
}

export interface GoogleLoginRequest {
  idToken: string;
}

export interface ExternalLoginRequest {
  provider: string;
  idToken: string;
}

export interface RefreshTokenRequest {
  refreshToken: string;
}

export interface LogoutRequest {
  refreshToken: string | null;
}

export interface UpdateProfileRequest {
  displayName: string | null;
  username: string | null;
  role: string | null;
  team: string | null;
  bio: string | null;
}
